compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -360,6 +365,28 @@ class TypeMismatch(val found: Type, expected: Type, val inTree: Option[untpd.Tre
     val treeStr = inTree.map(x => s"\nTree:\n\n${x.show}\n").getOrElse("")
     treeStr + "\n" + super.explain
 
+  override def actions(using Context) =
+    if shouldSuggestNN then
+      inTree match {
+        case Some(tree) if tree != null =>
+          val content = tree.source.content().slice(tree.srcPos.startPos.start, tree.srcPos.endPos.end).mkString
+          val replacement = tree match
+            case Apply(fun, args) => "(" + content + ").nn"
+            case _ => content + ".nn"
+          List(
+            CodeAction(title = """Add .nn""",
+            description = None,
+            patches = List(
+              ActionPatch(tree.srcPos.sourcePos, replacement)
+            )
+            )
+          )
+        case _ =>
+          List()
+      }
+    else
```

Comment:
We can get rid of the outter `if` by moving `shouldSuggestNN` to the match case condition.